src/pdf_beamtime/src/pdf_beamtime_fidpose_server.cpp

```diff
@@ -0,0 +1,626 @@
+/*Copyright 2024 Brookhaven National Laboratory
+BSD 3 Clause License. See LICENSE.txt for details.*/
+#include <pdf_beamtime/pdf_beamtime_fidpose_server.hpp>
+
+using namespace std::placeholders;
+
+PdfBeamtimeFidPoseServer::PdfBeamtimeFidPoseServer(
+  const std::string & move_group_name = "ur_manipulator",
+  const rclcpp::NodeOptions & options = rclcpp::NodeOptions(),
+  std::string action_name = "pdf_beamtime_fidpose_action_server")
+: PdfBeamtimeServer(move_group_name, options, "pdf_beamtime_action_server")
+{
+  tf_utilities_ = new TFUtilities(node_);
+
+  // Create the action server
+  fidpose_action_server_ = rclcpp_action::create_server<FidPoseControlMsg>(
+    this->node_,
+    action_name,
+    std::bind(&PdfBeamtimeFidPoseServer::fidpose_handle_goal, this, _1, _2),
+    std::bind(&PdfBeamtimeFidPoseServer::fidpose_handle_cancel, this, _1),
+    std::bind(&PdfBeamtimeFidPoseServer::fidpose_handle_accepted, this, _1));
+}
+
+rclcpp_action::GoalResponse PdfBeamtimeFidPoseServer::fidpose_handle_goal(
+  const rclcpp_action::GoalUUID & uuid,
+  std::shared_ptr<const FidPoseControlMsg::Goal> goal)
+{
+  (void)uuid;
+  return rclcpp_action::GoalResponse::ACCEPT_AND_EXECUTE;
+}
```

Comment:
This is where I would expect some logic on whether or not we have an existing position to return to?

Reply:
Logic added to reject if  a return goal is issued before return positions are saved
https://github.com/maffettone/erobs/blob/a54c58888c733d986f5fe443e0b3d04d33e9c634/src/pdf_beamtime/src/pdf_beamtime_fidpose_server.cpp#L29-L33